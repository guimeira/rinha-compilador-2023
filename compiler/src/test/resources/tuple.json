{"name":"tuple.rinha","expression":{"kind":"Let","name":{"text":"a","location":{"start":4,"end":5,"filename":"tuple.rinha"}},"value":{"kind":"Tuple","first":{"kind":"Int","value":1,"location":{"start":9,"end":10,"filename":"tuple.rinha"}},"second":{"kind":"Str","value":"str","location":{"start":12,"end":17,"filename":"tuple.rinha"}},"location":{"start":8,"end":18,"filename":"tuple.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":24,"end":25,"filename":"tuple.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"a","location":{"start":34,"end":35,"filename":"tuple.rinha"}},"location":{"start":28,"end":36,"filename":"tuple.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":42,"end":43,"filename":"tuple.rinha"}},"value":{"kind":"Print","value":{"kind":"First","value":{"kind":"Var","text":"a","location":{"start":58,"end":59,"filename":"tuple.rinha"}},"location":{"start":52,"end":60,"filename":"tuple.rinha"}},"location":{"start":46,"end":61,"filename":"tuple.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":67,"end":68,"filename":"tuple.rinha"}},"value":{"kind":"Print","value":{"kind":"Second","value":{"kind":"Var","text":"a","location":{"start":84,"end":85,"filename":"tuple.rinha"}},"location":{"start":77,"end":86,"filename":"tuple.rinha"}},"location":{"start":71,"end":87,"filename":"tuple.rinha"}},"next":{"kind":"Let","name":{"text":"b","location":{"start":93,"end":94,"filename":"tuple.rinha"}},"value":{"kind":"Tuple","first":{"kind":"Tuple","first":{"kind":"Tuple","first":{"kind":"Int","value":1,"location":{"start":100,"end":101,"filename":"tuple.rinha"}},"second":{"kind":"Int","value":2,"location":{"start":102,"end":103,"filename":"tuple.rinha"}},"location":{"start":99,"end":104,"filename":"tuple.rinha"}},"second":{"kind":"Tuple","first":{"kind":"Int","value":10,"location":{"start":106,"end":108,"filename":"tuple.rinha"}},"second":{"kind":"Int","value":20,"location":{"start":109,"end":111,"filename":"tuple.rinha"}},"location":{"start":105,"end":112,"filename":"tuple.rinha"}},"location":{"start":98,"end":113,"filename":"tuple.rinha"}},"second":{"kind":"Tuple","first":{"kind":"Tuple","first":{"kind":"Str","value":"a","location":{"start":116,"end":119,"filename":"tuple.rinha"}},"second":{"kind":"Str","value":"b","location":{"start":120,"end":123,"filename":"tuple.rinha"}},"location":{"start":115,"end":124,"filename":"tuple.rinha"}},"second":{"kind":"Tuple","first":{"kind":"Str","value":"A","location":{"start":126,"end":129,"filename":"tuple.rinha"}},"second":{"kind":"Str","value":"B","location":{"start":130,"end":133,"filename":"tuple.rinha"}},"location":{"start":125,"end":134,"filename":"tuple.rinha"}},"location":{"start":114,"end":135,"filename":"tuple.rinha"}},"location":{"start":97,"end":136,"filename":"tuple.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":142,"end":143,"filename":"tuple.rinha"}},"value":{"kind":"Print","value":{"kind":"First","value":{"kind":"First","value":{"kind":"First","value":{"kind":"Var","text":"b","location":{"start":170,"end":171,"filename":"tuple.rinha"}},"location":{"start":164,"end":172,"filename":"tuple.rinha"}},"location":{"start":158,"end":173,"filename":"tuple.rinha"}},"location":{"start":152,"end":174,"filename":"tuple.rinha"}},"location":{"start":146,"end":175,"filename":"tuple.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":181,"end":182,"filename":"tuple.rinha"}},"value":{"kind":"Print","value":{"kind":"Second","value":{"kind":"First","value":{"kind":"First","value":{"kind":"Var","text":"b","location":{"start":210,"end":211,"filename":"tuple.rinha"}},"location":{"start":204,"end":212,"filename":"tuple.rinha"}},"location":{"start":198,"end":213,"filename":"tuple.rinha"}},"location":{"start":191,"end":214,"filename":"tuple.rinha"}},"location":{"start":185,"end":215,"filename":"tuple.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":221,"end":222,"filename":"tuple.rinha"}},"value":{"kind":"Print","value":{"kind":"First","value":{"kind":"Second","value":{"kind":"First","value":{"kind":"Var","text":"b","location":{"start":250,"end":251,"filename":"tuple.rinha"}},"location":{"start":244,"end":252,"filename":"tuple.rinha"}},"location":{"start":237,"end":253,"filename":"tuple.rinha"}},"location":{"start":231,"end":254,"filename":"tuple.rinha"}},"location":{"start":225,"end":255,"filename":"tuple.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":261,"end":262,"filename":"tuple.rinha"}},"value":{"kind":"Print","value":{"kind":"Second","value":{"kind":"Second","value":{"kind":"First","value":{"kind":"Var","text":"b","location":{"start":291,"end":292,"filename":"tuple.rinha"}},"location":{"start":285,"end":293,"filename":"tuple.rinha"}},"location":{"start":278,"end":294,"filename":"tuple.rinha"}},"location":{"start":271,"end":295,"filename":"tuple.rinha"}},"location":{"start":265,"end":296,"filename":"tuple.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":302,"end":303,"filename":"tuple.rinha"}},"value":{"kind":"Print","value":{"kind":"First","value":{"kind":"First","value":{"kind":"Second","value":{"kind":"Var","text":"b","location":{"start":331,"end":332,"filename":"tuple.rinha"}},"location":{"start":324,"end":333,"filename":"tuple.rinha"}},"location":{"start":318,"end":334,"filename":"tuple.rinha"}},"location":{"start":312,"end":335,"filename":"tuple.rinha"}},"location":{"start":306,"end":336,"filename":"tuple.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":342,"end":343,"filename":"tuple.rinha"}},"value":{"kind":"Print","value":{"kind":"Second","value":{"kind":"First","value":{"kind":"Second","value":{"kind":"Var","text":"b","location":{"start":372,"end":373,"filename":"tuple.rinha"}},"location":{"start":365,"end":374,"filename":"tuple.rinha"}},"location":{"start":359,"end":375,"filename":"tuple.rinha"}},"location":{"start":352,"end":376,"filename":"tuple.rinha"}},"location":{"start":346,"end":377,"filename":"tuple.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":383,"end":384,"filename":"tuple.rinha"}},"value":{"kind":"Print","value":{"kind":"First","value":{"kind":"Second","value":{"kind":"Second","value":{"kind":"Var","text":"b","location":{"start":413,"end":414,"filename":"tuple.rinha"}},"location":{"start":406,"end":415,"filename":"tuple.rinha"}},"location":{"start":399,"end":416,"filename":"tuple.rinha"}},"location":{"start":393,"end":417,"filename":"tuple.rinha"}},"location":{"start":387,"end":418,"filename":"tuple.rinha"}},"next":{"kind":"Print","value":{"kind":"Second","value":{"kind":"Second","value":{"kind":"Second","value":{"kind":"Var","text":"b","location":{"start":447,"end":448,"filename":"tuple.rinha"}},"location":{"start":440,"end":449,"filename":"tuple.rinha"}},"location":{"start":433,"end":450,"filename":"tuple.rinha"}},"location":{"start":426,"end":451,"filename":"tuple.rinha"}},"location":{"start":420,"end":452,"filename":"tuple.rinha"}},"location":{"start":379,"end":452,"filename":"tuple.rinha"}},"location":{"start":338,"end":452,"filename":"tuple.rinha"}},"location":{"start":298,"end":452,"filename":"tuple.rinha"}},"location":{"start":257,"end":452,"filename":"tuple.rinha"}},"location":{"start":217,"end":452,"filename":"tuple.rinha"}},"location":{"start":177,"end":452,"filename":"tuple.rinha"}},"location":{"start":138,"end":452,"filename":"tuple.rinha"}},"location":{"start":89,"end":452,"filename":"tuple.rinha"}},"location":{"start":63,"end":452,"filename":"tuple.rinha"}},"location":{"start":38,"end":452,"filename":"tuple.rinha"}},"location":{"start":20,"end":452,"filename":"tuple.rinha"}},"location":{"start":0,"end":452,"filename":"tuple.rinha"}},"location":{"start":0,"end":452,"filename":"tuple.rinha"}}
