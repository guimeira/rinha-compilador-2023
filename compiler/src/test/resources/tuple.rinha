let a = (1, "str");
let _ = print(a);
let _ = print(first(a));
let _ = print(second(a));
let b = (((1,2),(10,20)),(("a","b"),("A","B")));
let _ = print(first(first(first(b))));
let _ = print(second(first(first(b))));
let _ = print(first(second(first(b))));
let _ = print(second(second(first(b))));
let _ = print(first(first(second(b))));
let _ = print(second(first(second(b))));
let _ = print(first(second(second(b))));
print(second(second(second(b))))
