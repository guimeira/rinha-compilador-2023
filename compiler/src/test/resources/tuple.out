(1,str)
1
str
1
2
10
20
a
b
A
B
